core/jvmMain/src/kotlinx/io/buffer/BufferOperations.kt

```diff
@@ -159,3 +158,23 @@ actual inline fun <R> ByteArray.useBuffer(offset: Int, length: Int, block: (Buff
             .slice().order(ByteOrder.BIG_ENDIAN)
     ).let(block)
 }
+
+/**
+ * Compacts the [Buffer]. Move content from ([startIndex], [endIndex]) range to (0, 'endIndex - startIndex') range.
+ * The copying ranges can overlap.
+ *
+ * @return [endIndex] - [startIndex] (copied bytes count) or updated [endIndex]
+ */
+internal actual fun Buffer.compact(startIndex: Int, endIndex: Int): Int {
```

Comment:
Could you please elaborate what's the point of return value?
Why is it useful?

Reply:
It's the new end index, it's easy to combine:
```kotlin
endIndex = buffer.compact(startIndex, endIndex)
```
instead of:
```kotlin
buffer.compact(startIndex, endIndex)
endIndex -= startIndex
```

Reply:
Good for internal use